xwayland/xwm.c

```diff
@@ -1518,7 +1543,17 @@ void wlr_xwayland_surface_restack(struct wlr_xwayland_surface *surface,
 	}
 	values[idx++] = mode;
 
-	xcb_configure_window(xwm->xcb_conn, surface->window_id, flags, values);
+	xcb_configure_window(xwm->xcb_conn, xsurface->window_id, flags, values);
+
+	if (mode == XCB_STACK_MODE_ABOVE) {
+		wl_list_remove(&xsurface->focus_link);
+		wl_list_insert(&xwm->surfaces_in_focus_order, &xsurface->focus_link);
+	} else {
+		// TODO
```

Comment:
/cc @ammen99: Sway doesn't use this function so I'm not sure what the right behavior here is (or how I'd exercise this code path to test). Thoughts?